File: python code/basic functions/cropandresize.py
import cv2

# reading the image
source = cv2.imread("../../images/lena.jpg", flags=1)

scaleX = 0.6
scaleY = 0.6

# scaling down the image 0.6 times
scaleDown = cv2.resize(source, None, fx=scaleX, fy=scaleY, interpolation=cv2.INTER_LINEAR)

# scaling up the image 1.8 times
scaleUp = cv2.resize(source, None, fx=scaleY * 3, fy=scaleY * 3, interpolation=cv2.INTER_LINEAR)

# crop image
crop = source[50:150, 20:200]

# display all the images
cv2.imshow("Original", source)
cv2.imshow("Scaled Down", scaleDown)
cv2.imshow("Scaled Up", scaleUp)
cv2.imshow("Cropped Image", crop)

cv2.waitKey(0)

File: c++ code/week1/examples/basic functions/basic functions/cropandresize.cpp
#include <opencv2/opencv.hpp>

int main2()
{
	cv::Mat source, scaleDown, scaleUp;

	// read souce image
	source = cv::imread("../../../../../images/lena.jpg");

	// scaling factors
	double scaleX = 0.6;
	double scaleY = 0.6;

	// scaling down the image 0.6 times
	cv::resize(source, scaleDown, cv::Size(), scaleX, scaleY, cv::INTER_LINEAR);
	// scaling up the image 1.8 times
	cv::resize(source, scaleUp, cv::Size(), scaleX * 3, scaleY * 3, cv::INTER_LINEAR);

	// cropped image
	cv::Mat crop = source(cv::Range(50, 150), cv::Range(20, 200));

	// create dsiplay windows for all three images
	cv::namedWindow("Original", cv::WINDOW_AUTOSIZE);
	cv::namedWindow("Scale Down", cv::WINDOW_AUTOSIZE);
	cv::namedWindow("Scale Up", cv::WINDOW_AUTOSIZE);
	cv::namedWindow("Croped Image", cv::WINDOW_AUTOSIZE);

	// show image
	cv::imshow("Original", source);
	cv::imshow("Scale Down", scaleDown);
	cv::imshow("Scale Up", scaleUp);
	cv::imshow("Cropped Image", crop);

	cv::waitKey(0);

	return EXIT_SUCCESS;
}